## Write a python program to print all natural numbers in reverse (from n to 1). – using while loop


In [37]:
n=int(input('Input N:'))
i=1
print(f'Natural numbers from {n} to 1 in reverse: ', end='')
while (i<=n):
    print(n,end=", ")
    n-=1    

Input N: 6


6, 5, 4, 3, 2, 1, Natural numbers from 0-1 in reverse: 

In [39]:
n=int(input("Input uper limit: "))
i=1
print(f"Natural numbers from 1 to {n}: ", end="")
while i<=n:
  print(i, end=",")
  i=i+1


Input uper limit:  5


Natural numbers from 1 to 5: 1,2,3,4,5,

In [4]:
#Write a C program to print all natural numbers in reverse (from n to 1). – using while loop
n=int(input("Input uper limit: "))
i=1
print(f"Natural numbers from 1 to {n}: ", end="")
while (i<=n):
  print(n,end="")
  if n>i:
      print(",", end="")
  n -=1
print()

Input uper limit:  54


Natural numbers from 1 to 54: 54,53,52,51,50,49,48,47,46,45,44,43,42,41,40,39,38,37,36,35,34,33,32,31,30,29,28,27,26,25,24,23,22,21,20,19,18,17,16,15,14,13,12,11,10,9,8,7,6,5,4,3,2,1


## QUS 03- Write a python program to print all alphabets from a to z. – using while loop

In [55]:
ch= 'a'
while ch <= 'z' :
    print(ch, end=" ")
    ch= chr(ord(ch) +1)



a b c d e f g h i j k l m n o p q r s t u v w x y z a b c d e f g h i j k l m n o p q r s t u v w x y z 

In [2]:
# Write a python program to print all alphabets from a to z. – using while loop

alphabet = ord('a') # Get the ASCII value of 'a'
while alphabet <= ord('z'): # Loop until the ASCII value of 'z' is reached
  print(chr(alphabet), end=" ") # Print the character corresponding to the ASCII value
  alphabet += 1 # Increment the ASCII value to get the next character

a b c d e f g h i j k l m n o p q r s t u v w x y z 

## using for loop

In [17]:
for letter in range(ord("a"),ord("z")+1):
    print(chr(letter), end=" ")

a b c d e f g h i j k l m n o p q r s t u v w x y z 

## QUS 04- Write a python program to print all even numbers between 1 to 100. – using while loop

In [22]:
n=100
i=1
print("Even numbers between 1 to 100 :",end="")
while (i<=n):
    if i%2==0 :
        print(i,end="")
        if (i<n):
            print(",",end="")
    i=i+1
    

Even numbers between 1 to 100 :2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94,96,98,100

## QUS 05- Write a python program to print all odd number between 1 to 100.

In [27]:
n=100
i=1
print('Odd number between 1 ton 100: ',end='')
while i<=n :
    if i%2==1:
        print(i,end="")
        if i<99:
            print(',',end="")
    i=i+1

Odd number between 1 ton 100: 1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95,97,99

## QUS 06- Write a Python program to find sum of all natural numbers between 1 to n.

In [38]:
n=int(input("input upprt limit: "))
i=1
s=0
print(f'Sum of the natural number 1 to {n}: ',end='')
while i<=n:
    s= s+i
    i=i+1
print(s,end='')

input upprt limit:  5


Sum of the natural number 1 to 5: 15

## QUS 07- Write a Python program to find sum of all even numbers between 1 to n.

In [53]:
n=int(input("Input upper limit of even number:"))
i=1
s=0
while i<=n :
    if i%2==0 :
        s= s+i
    i=i+1
print(f'Sum of even numbers between 1 to {n}:',s)
            

Input upper limit of even number: 11


Sum of even numbers between 1 to 11: 30


## Using for loop

In [70]:
n=int(input("Input upper limit of even number:"))
s=0
for i in range(1,n+1,1):
  if i%2==0:
      s=s+i
print(f'Sum of even numbers between 1 to {n}:',s)


Input upper limit of even number: 10


Sum of even numbers between 1 to 10: 30


## QUS 08-Write a Python program to find sum of all odd numbers between 1 to n.

In [75]:
n=int(input("Input upper limit of odd number:"))
i=1
s=0
while i<=n :
    if i%2==1 :
        s= s+i
    i=i+1
print(f'Sum of odd numbers between 1 to {n}:',s)

Input upper limit of odd number: 10


Sum of odd numbers between 1 to 10: 25


In [77]:
# using for loop
n=int(input("Input upper limit of even number:"))
s=0
for i in range(1,n+1,1):
  if i%2==1:
      s=s+i
print(f'Sum of even numbers between 1 to {n}:',s)

Input upper limit of even number: 10


Sum of even numbers between 1 to 10: 25


## QUS 09- Write a python program to print multiplication table of any number.

In [80]:
n= int(input("Input number: "))
i=1
while i<=10 :
    m=n*i
    print(f'{n}*{i}={m}')
    i=i+1

Input number:  5


5*1=5
5*2=10
5*3=15
5*4=20
5*5=25
5*6=30
5*7=35
5*8=40
5*9=45
5*10=50


In [84]:
# using for loop
n= int(input("Input number: "))
for i in range(1,10,1):
    m=n*i
    print(f'{n}*{i}={m}')

Input number:  5


5*1=5
5*2=10
5*3=15
5*4=20
5*5=25
5*6=30
5*7=35
5*8=40
5*9=45


# QUS 10- Write a Python program to count number of digits in a number.

In [141]:
num=int(input('Input number: '))
count=0
num=abs(num)
while num>0:
    num//= 10
    count= count+1
        
print(f"Number of digit:{count}")
    

Input number:  -9876


Number of digit:4


In [163]:
num=input('Input number: ')
count=0

for digit in num:
    if digit.isdigit():
        count += 1
print(f"Number of digit:{count}")

Input number:  56


Number of digit:2


## QUS 11- Write a Python program to find first and last digit of a number.

In [7]:
num=int(input("Input number"))
last_digit=num%10
fast_digit=num
while fast_digit >=10:
    fast_digit//=10
print(f"First digit: {fast_digit}")
print("Last digit:",last_digit)

Input number 34567


First digit: 3
Last digit: 7


# QUS 12-Write a Python program to find sum of first and last digit of a number.

In [12]:
num=int(input("Input number"))
last_digit=num%10
fast_digit=num
while fast_digit >=10:
    fast_digit//=10
sum=fast_digit+last_digit
print(f"Sum of first and last digit: {sum}")


Input number 1234


Sum of first and last digit: 5


# QUS 13-Write a Python program to swap first and last digits of a number.

In [14]:
num=int(input("Input number"))
n=num
count= 0
while n>0:
    n//=10
    count +=1
last_digit= num%10
first_digit = num // (10**(count - 1))
middle = (num % (10**(count-1)))//10
swapped_num = (last_digit * (10** (count-1)))+(middle * 10)+ first_digit
print("Number after swapping first and last digits: ", swapped_num)

Input number 12345


Number after swapping first and last digits:  52341


## QUS 14-Write a python program to calculate sum of digits of a number.

In [4]:
# Write a python program to calculate sum of digits of a number.

num = int(input("Enter a number: "))
sum_digits = 0

# Use a while loop to extract and sum the digits
while num > 0:
  digit = num % 10  # Get the last digit
  sum_digits += digit  # Add the digit to the sum
  num //= 10  # Remove the last digit from the number

print(f"The sum of the digits is: {sum_digits}")

Enter a number:  1234


The sum of the digits is: 10


## QUS 15- Write a Python program to calculate product of digits of a number.

In [17]:
# Write a python program to calculate product of digits of a number.

num = int(input("Enter a number: "))
product_digits = 1

# Use a while loop to extract and multiply the digits
while num > 0:
  digit = num % 10  # Get the last digit
  product_digits *= digit  # Multiply the product by the digit
  num //= 10  # Remove the last digit from the number

print(f"The product of the digits is: {product_digits}")


Enter a number:  23


The product of the digits is: 6


## QUS 16- Write a Python program to enter a number and print its reverse.

In [20]:
# Get input from the user
number = input("Enter a number: ")

# Reverse the number using string slicing
reversed_number = number[::-1]

# Print the reversed number
print("Reversed number:", reversed_number)

Enter a number:  12345


Reversed number: 54321


In [28]:
# Write a Python program to enter a number and print its reverse while loop.

num = int(input("Enter a number: "))
reversed_num = 0

# Use a while loop to reverse the number
while num > 0:
  digit = num % 10  # Get the last digit
  reversed_num = (reversed_num * 10) + digit  # Add the digit to the reversed number
  num //= 10  # Remove the last digit from the number

print(f"The reversed number is: {reversed_num}")

Enter a number:  123


The reversed number is: 321


In [26]:
# using for loop
# Get input from the user
number = input("Enter a number: ")

# Initialize an empty string for the reversed number
reversed_number = ""

# Use a for loop to iterate through each digit in reverse order
for digit in range(len(number)-1, -1, -1):
    reversed_number += number[digit]

# Print the reversed number
print("Reversed number:", reversed_number)

Enter a number:  12345


Reversed number: 54321


## QUS 17- Write a Python program to check whether a number is palindrome or not.

In [35]:
num=input("input number:")
reverse_num=num[::-1]
if reverse_num ==num:
    print(f'{num} is palindrome')
else:
    print(f'{num} is not a palindrome')

          

input number: 1212121


1212121 is palindrome


### Using while loop

In [37]:
# Write a Python program to check whether a number is palindrome or not.

num = int(input("Enter a number: "))
reversed_num = 0

# Reverse the number
while num > 0:
  digit = num % 10
  reversed_num = (reversed_num * 10) + digit
  num //= 10

# Check if the original number is equal to the reversed number
if num == reversed_num:
  print(f"{num} is a palindrome number.")
else:
  print(f"{num} is not a palindrome number.")

Enter a number:  1234


0 is not a palindrome number.


## QUS 18-Write a python program to find frequency of each digit in a given integer.

In [29]:
#using for loop
def find_digit_frequency(number):
    number = int(input("Enter an integer: "))
    # Convert the number to a string to iterate through its digits
    num_str = str(abs(number))
    
    # Create a dictionary to store the frequency of each digit
    frequency = {}
    
    # Count the frequency of each digit
    for digit in num_str:
        if digit in frequency:
            frequency[digit] += 1
        else:
            frequency[digit] = 1
    
    return frequency

# Example usage
result = find_digit_frequency(number)

print("Digit frequencies:")
for digit, count in sorted(result.items()):
    print(f"Digit {digit}: {count} times")

Enter an integer:  00000101


Digit frequencies:
Digit 0: 1 times
Digit 1: 2 times


In [27]:
# using while looop
# Write a python program to find frequency of each digit in a given integer.

num = int(input("Enter a number: "))
temp = abs(num)
digit_counts = {}

# Use a while loop to count 
# Special case for 0
if num == 0:
    digit_counts[0] = 1
while temp > 0:
    digit = temp % 10
    if digit in digit_counts:
        digit_counts[digit] += 1
    else:
        digit_counts[digit] = 1
    temp //= 10

print(f"Frequency of each digit in {num}:")
for digit, count in digit_counts.items():
    print(f"Digit {digit}: {count}")

Enter a number:  01010


Frequency of each digit in 1010:
Digit 0: 2
Digit 1: 2


In [41]:
#using for loop (count 0)
num = input("Enter a number: ")  # Get input as string to preserve leading zeros
digit_counts = {}

# Count frequency of each digit
for digit in num:
    if digit == '-':  # Skip negative sign
        continue
    digit = int(digit)  # Convert character to integer
    if digit in digit_counts:
        digit_counts[digit] += 1
    else:
        digit_counts[digit] = 1

print(f"Frequency of each digit in {num}:")
for digit, count in sorted(digit_counts.items()):
    print(f"Digit {digit}: {count}")

Enter a number:  64536


Frequency of each digit in 64536:
Digit 3: 1
Digit 4: 1
Digit 5: 1
Digit 6: 2


## বাংলা ব্যাখ্যা (Explanation in Bangla):

এই নতুন কোডে আমি একটি ভিন্ন পদ্ধতি ব্যবহার করেছি:

১. **স্ট্রিং হিসেবে ইনপুট নেওয়া**: আমরা সংখ্যাকে স্ট্রিং হিসেবে ইনপুট নিচ্ছি, যাতে শুরুতে থাকা 0 গুলোও সংরক্ষিত থাকে।

২. **প্রতিটি অক্ষর পরীক্ষা**: আমরা for লুপ ব্যবহার করে স্ট্রিং-এর প্রতিটি অক্ষর পরীক্ষা করছি।

৩. **ঋণাত্মক চিহ্ন এড়িয়ে যাওয়া**: যদি অক্ষরটি '-' হয় (ঋণাত্মক সংখ্যার জন্য), তাহলে আমরা এটি এড়িয়ে যাই।

৪. **অঙ্ক গণনা**: প্রতিটি অক্ষরকে সংখ্যায় রূপান্তর করে আমরা ডিকশনারিতে তার ফ্রিকোয়েন্সি বাড়াই।

এই পদ্ধতির সুবিধা হল:
- এটি সংখ্যার সব অবস্থানে থাকা 0 গুলো সঠিকভাবে গণনা করে
- শুরুতে থাকা 0 গুলোও গণনা করা হয় (যেমন "0123" এ প্রথম 0)
- ঋণাত্মক সংখ্যাও সঠিকভাবে হ্যান্ডেল করা হয়

উদাহরণস্বরূপ:
- যদি ইনপুট হয় `1001`, আউটপুট হবে: "Digit 0: 2, Digit 1: 2"
- যদি ইনপুট হয় `0123`, আউটপুট হবে: "Digit 0: 1, Digit 1: 1, Digit 2: 1, Digit 3: 1"
- যদি ইনপুট হয় `-101`, আউটপুট হবে: "Digit 0: 1, Digit 1: 2"

এই কোডটি সব ধরনের সংখ্যার জন্য সঠিকভাবে কাজ করবে এবং সব অবস্থানে থাকা 0 গুলো সঠিকভাবে গণনা করবে।

In [49]:
# using while loop
num = input("Enter a number: ")  # Get input as string to preserve all digits
digit_counts = {}

# Use while loop to count the frequency of each digit
i = 0
while i < len(num):
    if num[i] == '-':  # Skip negative sign
        i += 1
        continue
    
    digit = int(num[i])  # Convert character to integer
    if digit in digit_counts:
        digit_counts[digit] += 1
    else:
        digit_counts[digit] = 1
    i += 1

print(f"Frequency of each digit in {num}:")
for digit, count in sorted(digit_counts.items()):
    print(f"Digit {digit}: {count}")

Enter a number:  00455


Frequency of each digit in 00455:
Digit 0: 2
Digit 4: 1
Digit 5: 2


## বাংলা ব্যাখ্যা (Explanation in Bangla):

এই কোডে আমি while লুপ ব্যবহার করে সমস্যাটি সমাধান করেছি:

১. **স্ট্রিং হিসেবে ইনপুট নেওয়া**: আমরা সংখ্যাকে স্ট্রিং হিসেবে ইনপুট নিচ্ছি, যাতে সব অঙ্ক (শুরু, মাঝে বা শেষে থাকা 0 সহ) সংরক্ষিত থাকে।

২. **while লুপ ব্যবহার**: আমরা একটি কাউন্টার `i` ব্যবহার করে স্ট্রিং-এর প্রতিটি অক্ষর পর্যালোচনা করছি। লুপটি তখন পর্যন্ত চলবে যতক্ষণ না `i` স্ট্রিং-এর দৈর্ঘ্য অতিক্রম করে।

৩. **ঋণাত্মক চিহ্ন এড়িয়ে যাওয়া**: যদি অক্ষরটি '-' হয় (ঋণাত্মক সংখ্যার জন্য), তাহলে আমরা এটি এড়িয়ে যাই।

৪. **অঙ্ক গণনা**: প্রতিটি অক্ষরকে সংখ্যায় রূপান্তর করে আমরা ডিকশনারিতে তার ফ্রিকোয়েন্সি বাড়াই।

৫. **কাউন্টার বৃদ্ধি**: প্রতিটি ইটারেশনের শেষে আমরা `i` এর মান 1 বাড়াই।

এই পদ্ধতির সুবিধা হল:
- এটি while লুপ ব্যবহার করে, যা আপনার প্রয়োজন অনুযায়ী
- সংখ্যার সব অবস্থানে থাকা 0 গুলো সঠিকভাবে গণনা করে
- শুরুতে থাকা 0 গুলোও গণনা করা হয় (যেমন "0123" এ প্রথম 0)
- ঋণাত্মক সংখ্যাও সঠিকভাবে হ্যান্ডেল করা হয়

উদাহরণস্বরূপ:
- যদি ইনপুট হয় `1001`, আউটপুট হবে: "Digit 0: 2, Digit 1: 2"
- যদি ইনপুট হয় `0123`, আউটপুট হবে: "Digit 0: 1, Digit 1: 1, Digit 2: 1, Digit 3: 1"
- যদি ইনপুট হয় `-101`, আউটপুট হবে: "Digit 0: 1, Digit 1: 2"

এই কোডটি while লুপ ব্যবহার করে সব ধরনের সংখ্যার জন্য সঠিকভাবে কাজ করবে এবং সব অবস্থানে থাকা 0 গুলো সঠিকভাবে গণনা করবে।

# QUS 19-Write a PYTHON program to enter a number and print it in words.

In [54]:
def number_to_words(num):
    # Dictionary for numbers 0-19
    ones = {
        0: 'Zero', 1: 'One', 2: 'Two', 3: 'Three', 4: 'Four', 
        5: 'Five', 6: 'Six', 7: 'Seven', 8: 'Eight', 9: 'Nine',
        10: 'Ten', 11: 'Eleven', 12: 'Twelve', 13: 'Thirteen', 14: 'Fourteen',
        15: 'Fifteen', 16: 'Sixteen', 17: 'Seventeen', 18: 'Eighteen', 19: 'Nineteen'
    }
    
    # Dictionary for tens place (20-90)
    tens = {
        2: 'Twenty', 3: 'Thirty', 4: 'Forty', 5: 'Fifty',
        6: 'Sixty', 7: 'Seventy', 8: 'Eighty', 9: 'Ninety'
    }
    
    # For numbers beyond 999, we'll limit to billions for simplicity
    scales = ['', 'Thousand', 'Million', 'Billion']
    
    # Special case for zero
    if num == 0:
        return ones[0]
    
    # For negative numbers
    if num < 0:
        return "Negative " + number_to_words(abs(num))
    
    words = ""
    
    # Process the number in groups of 3 digits
    scale_index = 0
    while num > 0:
        # Extract the last 3 digits
        group = num % 1000
        
        if group != 0:
            group_words = ""
            
            # Process hundreds place
            if group // 100 > 0:
                group_words += ones[group // 100] + " Hundred "
                group %= 100
            
            # Process tens and ones place
            if group > 0:
                # For numbers 1-19
                if group < 20:
                    group_words += ones[group]
                else:
                    # For numbers 20-99
                    group_words += tens[group // 10]
                    if group % 10 > 0:
                        group_words += " " + ones[group % 10]
            
            # Add the scale (thousand, million, etc.)
            if scale_index > 0:
                group_words += " " + scales[scale_index]
            
            # Add to the final result
            words = group_words + " " + words
        
        # Move to the next group of 3 digits
        num //= 1000
        scale_index += 1
    
    return words.strip()

# Get input from user
try:
    number = int(input("Enter a number: "))
    result = number_to_words(number)
    print(f"In words: {result}")
except ValueError:
    print("Please enter a valid integer.")

Enter a number:  34897536845


In words: Thirty Four Billion Eight Hundred Ninety Seven Million Five Hundred Thirty Six Thousand Eight Hundred Forty Five


In [80]:
def digit_to_word(digit):
    """Convert a single digit to its word representation."""
    words = {
        '0': 'Zero',
        '1': 'One',
        '2': 'Two',
        '3': 'Three',
        '4': 'Four',
        '5': 'Five',
        '6': 'Six',
        '7': 'Seven',
        '8': 'Eight',
        '9': 'Nine'
    }
    return words.get(digit, "")

def number_to_words_by_digit(number):
    """Convert each digit of a number to words."""
    # Convert number to string to process each digit
    num_str = str(number)
    
    # Convert each digit to word
    result = []
    for digit in num_str:
        result.append(digit_to_word(digit))
    
    # Join the words with spaces
    return " ".join(result)

# Get input from user
try:
    number = int(input("Enter a number: "))
    if number < 0:
        print("Negative", number_to_words_by_digit(abs(number)))
    else:
        print(number_to_words_by_digit(number))
except ValueError:
    print("Please enter a valid integer")

Enter a number:  -757456


Negative Seven Five Seven Four Five Six


In [74]:
# Get input from the user
number = int(input("Enter a number: "))
num1=str(abs(number))

# Reverse the number using string slicing
reversed_number = num1[::-1]
num = int(reversed_number)

while num > 0:
    digit = num % 10  # Get the last digit
    word=digit 
    if word==0:
        print("Zero",end=" ")
    elif word==1:
        print("One",end=" ")
    elif word==2:
        print("Two",end=" ")
    elif word==3:
        print("three",end=" ")
    elif word==4:
        print("Four",end=" ")
    elif word==5:
        print("Five",end=" ")
    elif word==6:
        print("Six",end=" ")
    elif word==7:
        print("Seven",end=" ")
    elif word==8:
        print("Eight",end=" ")
    elif word==9:
        print("Nine",end=" ")

    else:
        print("Plz enter valid number.")
    
    num //= 10  


Enter a number:  -87423


Eight Seven Four Two three 

# QUS 20- Write a PYTHON program to print all ASCII character with their values.

In [1]:
# Write a PYTHON program to print all ASCII character with their values.

# Iterate through ASCII values from 0 to 127 (standard ASCII range)
for value in range(128):
  character = chr(value)  # Convert the ASCII value to a character
  print(f"ASCII value: {value}, Character: {character}")

ASCII value: 0, Character: 
ASCII value: 1, Character: 
ASCII value: 2, Character: 
ASCII value: 3, Character: 
ASCII value: 4, Character: 
ASCII value: 5, Character: 
ASCII value: 6, Character: 
ASCII value: 7, Character: 
ASCII value: 8, Character:
ASCII value: 9, Character: 	
ASCII value: 10, Character: 

ASCII value: 11, Character: 
ASCII value: 12, Character: 
ASCII value: 13, Character: 
ASCII value: 14, Character: 
ASCII value: 15, Character: 
ASCII value: 16, Character: 
ASCII value: 17, Character: 
ASCII value: 18, Character: 
ASCII value: 19, Character: 
ASCII value: 20, Character: 
ASCII value: 21, Character: 
ASCII value: 22, Character: 
ASCII value: 23, Character: 
ASCII value: 24, Character: 
ASCII value: 25, Character: 
ASCII value: 26, Character: 
ASCII value: 27, Character: 
ASCII value: 28, Character: 
ASCII value: 29, Character: 
ASCII value: 30, Character: 
ASCII value: 31, Character: 
ASCII value: 32, Character:  
ASCII value: 33, Charac

# QUS 21-Write a Python program to find power of a number using for loop

In [26]:
x=int(input("Input base: "))
y=int(input("Input Exponent: "))
pow_of_num=1
for i in range (1,y+1,1):
    pow_of_num=pow_of_num*x
    
print(f"{x}^{y}={pow_of_num}")

Input base:  2
Input Exponent:  5


2^5=32


# QUS 22-Write a Python program to find all factors of a number.

In [64]:
x=int(input("Input Number: "))
print(f'factor of {x}: ',end="")
for i in range(1,x+1,1):
    if x%i==0:
        print(i,end="")
        if i<x:
            print("",end=",")


Input Number:  48


factor of 48: 1,2,3,4,6,8,12,16,24,48

# QUS 23-Write a Python program to calculate factorial of a number.

In [90]:
# Using for loop
x=int(input("Input Number: "))
fact=1
for i in range (1,x+1,1):
    fact=fact*i
    
print(f"factorial={fact}")

Input Number:  1


factorial=1


In [104]:
#using while loop
x=int(input("Input Number:"))
fact=1
factorial=1
while fact<=x:
    factorial=factorial*fact
    fact +=1
print(f'Factorial:{factorial}')

Input Number:  6


Factorial:720


# QUS 24-Write a Python program to find HCF (GCD) of two numbers.

In [145]:
# Find GCD using Euclidean algorithm
x = int(input("Input Number: "))
y = int(input("Input Number: "))
def find_gcd(a, b):
    while b:
        a, b = b, a % b
    return a

gcd = find_gcd(x, y)

print(f"The GCD of {x} and {y} is: {gcd}")

Input Number:  12
Input Number:  30


The GCD of 12 and 30 is: 6


# QUS 25-Write a python program to find LCM of two numbers.

In [13]:
def find_lcm(a, b):
    # LCM = (a * b) / GCD(a, b)
    return (a * b) // find_gcd(a, b)

x = int(input("Input 1st Number: "))
y = int(input("Input 2nd number: "))

lcm = find_lcm(x, y)

print(f"LCM: {lcm}")

Input 1st Number:  34
Input 2nd number:  43


LCM: 1462



## কোডের ব্যাখ্যা:

1. `find_gcd(a, b)` ফাংশনটি আগের মতোই ইউক্লিডিয়ান অ্যালগরিদম ব্যবহার করে GCD নির্ণয় করে।

2. `find_lcm(a, b)` ফাংশনটি দুটি সংখ্যার LCM নির্ণয় করে:
   - এটি সূত্র LCM(a, b) = (a * b) / GCD(a, b) ব্যবহার করে
   - `//` অপারেটর ব্যবহার করা হয়েছে যাতে ফলাফল পূর্ণসংখ্যা হয় (কারণ LCM সবসময় একটি পূর্ণসংখ্যা)

3. ব্যবহারকারীর কাছ থেকে দুটি সংখ্যা ইনপুট নেওয়া হয়।

4. `find_lcm` এবং `find_gcd` ফাংশন কল করে যথাক্রমে LCM এবং GCD নির্ণয় করা হয়।

5. সবশেষে, ফলাফল প্রিন্ট করা হয়।

এই কোডটি দুটি সংখ্যার মধ্যে GCD এবং LCM উভয়ই নির্ণয় করে, যা সংখ্যাতত্ত্বের বিভিন্ন সমস্যা সমাধানে উপযোগী।

# QUS 26-Write a Python program to check whether a number is Prime number or not.

In [48]:
# Using for loop
num = int(input("Input number:"))
is_prime = 1  # Flag to track if number is prime

for i in range(2, num):  # No need for step size 1 as it's the default
    if num % i == 0:
        print(f'{num} is not a prime number')
        is_prime = 0
        break  # Exit loop once we find a divisor

if is_prime and num > 1:  # 1 is not a prime number
    print(f'{num} is a prime number')

Input number: 43


43 is a prime number


# QUS 27- Write a Python program to print all Prime numbers between 1 to n.

In [ ]:
# Python program to print all prime numbers between 1 and n using only loops

# Get input from user
n = int(input("Enter the upper limit n: "))

print(f"Prime numbers between 1 and {n} are:")

# Check each number from 2 to n
for num in range(2, n+1):
    is_prime = True  # Assume number is prime initially
    
    # Check if num is divisible by any number from 2 to num-1
    for i in range(2, num):
        if num % i == 0:
            is_prime = False  # Not a prime number
            break
    
    # If is_prime is still True, print the number
    if is_prime:
        print(num, end=" ")

# QUS 28-Write a Python program to find sum of all prime numbers between 1 to n.

In [16]:
# Python program to print all prime numbers between 1 and n using only loops

# Get input from user
n = int(input("Enter the upper limit n: "))
summ=0

print(f"Prime numbers between 1 and {n} are:")

# Check each number from 2 to n
for num in range(2, n+1):
    is_prime = True  # Assume number is prime initially
    
    # Check if num is divisible by any number from 2 to num-1
    for i in range(2, num):
        if num % i == 0:
            is_prime = False  # Not a prime number
            break
    
    # If is_prime is still True, print the number
    if is_prime:
        summ=summ+num
        print(num, end=" ")
print("\n")
print("sum of the prime number is:",summ)


Enter the upper limit n:  10


Prime numbers between 1 and 10 are:
2 3 5 7 

sum of the prime number is: 17
